### PR TITLE
Add GitHub actions (from the inside)

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Description

#4088 rebased on latest master and from a branch in the official pylint repository so github actions work in the MR already see https://github.com/PyCQA/pylint/pull/4088#issuecomment-780180472

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :sparkles: New feature |

### CODE REVIEW

**Comment by cdce8p** [comment]
Already faster than Travis CI 🎉 I have to look at the coverage job again though as it seems.

@Pierre-Sassoulas Do you want me to update it? Maybe by pushing a new commit to #4088 that you can then cherry-pick?
I would also suggest to start removing parts of the travis and appveyor configs, eg. all formatting and pypy.

**Comment by Pierre-Sassoulas** [comment]
Yeah that would be very nice thank you for proposing. I'm trying to make coveralls work on one of my repo but I'm starting from scratch :)

**Comment by Pierre-Sassoulas** [comment]
Look like the solution: https://github.com/lemurheavy/coveralls-public/issues/1435#issuecomment-763357004

**Comment by cdce8p** [comment]
> Look like the solution: [lemurheavy/coveralls-public#1435 (comment)](https://github.com/lemurheavy/coveralls-public/issues/1435#issuecomment-763357004)

I'll have a look

**Comment by Pierre-Sassoulas** [comment]
Using `coveralls debug --service=gihtub` fixed it, (strangely not the same command without debug).

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+        run: |
+          . venv/bin/activate
+          coverage combine coverage*/.coverage
+          coverage report --rcfile=${{ env.COVERAGERC_FILE }}
+      - name: Upload coverage to Coveralls
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          . venv/bin/activate
+          coveralls debug --rcfile=${{ env.COVERAGERC_FILE }} --service=gihtub
```
Comment (cdce8p) [inline .github/workflows/ci.yaml:320]:
```suggestion
          coveralls debug --rcfile=${{ env.COVERAGERC_FILE }} --service=github
```

Comment (cdce8p) [reply .github/workflows/ci.yaml:320]: Now that is really strange. It works with `gihtub`?!

Comment (Pierre-Sassoulas) [reply .github/workflows/ci.yaml:320]: Seriously... I wonder how it works with this typo. I'm going to go sleep, I need it 😄

Comment (cdce8p) [reply .github/workflows/ci.yaml:320]: Maybe it just accepts the debug output as success?
The coveralls coverage comment is still missing which would suggest that it didn't work.

**Comment by cdce8p** [comment]
`docs`, `spelling` and `pypy` are also covered by Github Actions now.

**Comment by cdce8p** [comment]
I believe coveralls just worked 🎉

**Comment by cdce8p** [comment]
@Pierre-Sassoulas You can cherry-pick [this](https://github.com/cdce8p/pylint/pull/1/commits/a0a11c608a0d6012426162d2dd530f99a73b711c) commit here from my [branch](https://github.com/cdce8p/pylint/tree/github-actions). That should disable most of travis and appveyor (hopefully). Only travis deploy should would still work.

**Comment by cdce8p** [comment]
Just Appveyor now. By setting both `skip_tags: true` and `skip_non_tags: true` I likely caused it to not report anything. That's why the status is still on `Waiting`. I hope this commit might fix that: https://github.com/cdce8p/pylint/commit/2c14a4299bb5be217430aaa25df171ce9eb324bf. Instead of not running anything, my idea is to run `echo "Skip"`. That should always exit with `0` and thus the check should pass. Once Appveyor is deactivated, the `appveyor.yml` file can be removed completely.

**Comment by Pierre-Sassoulas** [comment]
I'm pushing the changes so we can see what they do without checking, but I think we'll have to really check that the new tests are doing the job of the old one before removing appveyor and travis (I put that in the 2.7.1 milestone because I'll wait for a review by at least maintainers).

**Comment by Pierre-Sassoulas** [comment]
I can already tell you that the caching you've done is very good, this will be a game changer for MR once it's merged (and we HAD to do it because Travis is changing its plan anyway).

**Comment by cdce8p** [comment]
Looks like everything is working now 🎉
Who would be able to do the review then? Form my point of view, this change could be does work and could be merged.

**Pull request merged**